File: tests/fixtures/sample.ts
import { readFileSync } from "fs";
import path from "path";

interface Processor {
    process(input: string): string;
}

class DataService implements Processor {
    private name: string;

    constructor(name: string) {
        this.name = name;
    }

    process(input: string): string {
        return this.formatOutput(input);
    }

    private formatOutput(value: string): string {
        return `${this.name}: ${value}`;
    }
}

class FileService {
    readFile(filePath: string): string {
        return readFileSync(filePath, "utf-8");
    }

    getBasename(filePath: string): string {
        return path.basename(filePath);
    }
}

function topLevelHelper(text: string): string {
    return text.trim();
}

const processText = (input: string): string => {
    return topLevelHelper(input);
};

File: tests/fixtures/sample.swift
import Foundation
import UIKit

protocol Processable {
    func process(input: String) -> String
}

class UserService: Processable {
    var name: String

    init(name: String) {
        self.name = name
    }

    func process(input: String) -> String {
        return formatOutput(input)
    }

    private func formatOutput(_ value: String) -> String {
        return "\(name): \(value)"
    }
}

struct Config {
    var debug: Bool
}

func topLevelHelper(text: String) -> String {
    return text.trimmingCharacters(in: .whitespaces)
}
